src/Microsoft.ML.AutoML/PipelineSuggesters/PipelineSuggester.cs

```diff
@@ -188,30 +191,39 @@ private static IValueGenerator[] ConvertToValueGenerators(IEnumerable<SweepableP
         /// Samples new hyperparameters for the trainer, and sets them.
         /// Returns true if success (new hyperparameters were suggested and set). Else, returns false.
         /// </summary>
-        private static bool SampleHyperparameters(MLContext context, SuggestedTrainer trainer, IEnumerable<SuggestedPipelineRunDetail> history, bool isMaximizingMetric)
+        private static bool SampleHyperparameters(MLContext context, SuggestedTrainer trainer,
+            IEnumerable<SuggestedPipelineRunDetail> history, bool isMaximizingMetric, IChannel logger)
         {
-            var sps = ConvertToValueGenerators(trainer.SweepParams);
-            var sweeper = new SmacSweeper(context,
-                new SmacSweeper.Arguments
+            try
+            {
+                var sps = ConvertToValueGenerators(trainer.SweepParams);
+                var sweeper = new SmacSweeper(context,
+                    new SmacSweeper.Arguments
+                    {
+                        SweptParameters = sps
+                    });
+
+                IEnumerable<SuggestedPipelineRunDetail> historyToUse = history
+                    .Where(r => r.RunSucceeded && r.Pipeline.Trainer.TrainerName == trainer.TrainerName && r.Pipeline.Trainer.HyperParamSet != null && r.Pipeline.Trainer.HyperParamSet.Any());
+
+                // get new set of hyperparameter values
+                var proposedParamSet = sweeper.ProposeSweeps(1, historyToUse.Select(h => h.ToRunResult(isMaximizingMetric))).First();
+                if (!proposedParamSet.Any())
                 {
-                    SweptParameters = sps
-                });
+                    return false;
+                }
 
-            IEnumerable<SuggestedPipelineRunDetail> historyToUse = history
-                .Where(r => r.RunSucceeded && r.Pipeline.Trainer.TrainerName == trainer.TrainerName && r.Pipeline.Trainer.HyperParamSet != null && r.Pipeline.Trainer.HyperParamSet.Any());
+                // associate proposed parameter set with trainer, so that smart hyperparameter
+                // sweepers (like KDO) can map them back.
+                trainer.SetHyperparamValues(proposedParamSet);
 
-            // get new set of hyperparameter values
-            var proposedParamSet = sweeper.ProposeSweeps(1, historyToUse.Select(h => h.ToRunResult(isMaximizingMetric))).First();
-            if (!proposedParamSet.Any())
+                return true;
+            }
+            catch (Exception ex)
             {
+                logger.Error($"SampleHyperparameters failed with exception: {ex}");
                 return false;
```

Comment:
The logging is useful, as it lets us write a more specific error message. But in the long term, I'm not sure if we should ignore the bugs in the sweeper. Catching here would make it difficult to locate bugs, like the internationalization bug which I think is causing this.

This specific bug is not just a non-deterministic bug occasionally occurring. It causes AutoML not work beyond 70 iterations for anyone in countries using a non-period decimal separator (e.g. "0,1234").

I might recommend [fixing the root bug](https://github.com/dotnet/machinelearning/pull/5163) instead of catching/ignoring here.

```suggestion
                throw; // Will re-throw the original exception with its original stack trace
```

Reply:
Thanks Justin, Yes, the culture bug is tracked at another PR as you mentioned. I re-throwed exception here as you suggested.

---
In reply to: [511428003](https://github.com/dotnet/machinelearning/pull/5455#discussion_r511428003) [](ancestors = 511428003)